File: src/main/java/com/cfgglobal/generator/entity/CodeField.java
package com.cfgglobal.generator.entity;

import lombok.Data;


@Data
public class CodeField {
    private Integer id;
    private String name;
    private String display;
    private Integer length;
    private Integer scale;
    private String type;
    private boolean isPrimaryKey;
    private boolean isSearchable;
    private boolean isSortable;
    private boolean isRequired;
    private boolean isUnique;
}

File: src/main/kotlin/com/cfgglobal/generator/entity/CodeField.kt
package com.cfgglobal.generator.entity


data class CodeField(
        val name: String,
        val display: String? = null,
        val type: FieldType,
        val primaryKey: Boolean = false,
        val searchable: Boolean = false,
        val sortable: Boolean = false,
        val required: Boolean = false,
        val unique: Boolean = false,
        val limit:Int = 255,
        val min: Long? = null,
        val max: Long? = null,
        val size: Size? = null


)

data class Size(
        val min: Int? = null,
        val max: Int? = null
)

